src/main/groovy/de.flachnetz.golang/GoPlugin.groovy

```diff
@@ -119,99 +126,105 @@ class GoPlugin implements Plugin<Project> {
 
-                if (!project.hasProperty("noDeps")) {
-                    if (new File(baseDir, "glide.yaml").exists()) {
-                        project.task("install-glide", type: Exec, dependsOn: "gopath") {
-                            commandLine go, "get", "github.com/Masterminds/glide"
-                            environment defaultEnvironmentVariables
-                        }
+                    project.task('clean-gopath', type: Exec) {
+                        commandLine "rm", "-rf", gopath
+                    }
 
-                        project.task("dependencies", type: Exec, dependsOn: "install-glide") {
-                            commandLine "${gopath}/bin/glide", "install", "--force", "--delete", "--cache"
-                            workingDir canonicalImportFile
-                            environment defaultEnvironmentVariables
-                        }
-                    } else {
-                        project.task("dependencies", type: Exec, dependsOn: "gopath") {
-                            commandLine go, "get", "-v"
-                            workingDir canonicalImportFile
-                            environment defaultEnvironmentVariables
-                        }
+                    project.task("install-goimports", type: Exec, dependsOn: ":gopath") {
+                        commandLine go, "get", "golang.org/x/tools/cmd/goimports"
+                        environment defaultEnvironmentVariables
                     }
-                }
 
-                project.task("install-goimports", type: Exec, dependsOn: "gopath") {
-                    commandLine go, "get", "golang.org/x/tools/cmd/goimports"
-                    environment defaultEnvironmentVariables
-                }
+                    project.task("install-junit-report", type: Exec, dependsOn: ":gopath") {
+                        commandLine go, "get", "github.com/jstemmer/go-junit-report"
+                        environment defaultEnvironmentVariables
+                    }
 
-                project.task("install-junit-report", type: Exec, dependsOn: "gopath") {
-                    commandLine go, "get", "github.com/jstemmer/go-junit-report"
-                    environment defaultEnvironmentVariables
-                }
+                    project.task("install-glide", type: Exec, dependsOn: ":gopath") {
+                        commandLine go, "get", "github.com/Masterminds/glide"
+                        environment defaultEnvironmentVariables
+                    }
 
-                project.task("optimize-imports", type: Exec, dependsOn: ["dependencies", "install-goimports"]) {
-                    commandLine(["${gopath}/bin/goimports", "-l", "-w"] + goSourceFiles)
-                    workingDir canonicalImportFile
-                    environment defaultEnvironmentVariables
-                }
+                    project.task("optimize-imports", type: Exec, dependsOn: ["dependencies", ":install-goimports"]) {
```

Comment:
Currently we need to put `optimize imports`, `format` and `test` into the root scope as we can not know to which packages those actions should apply to.